### PR TITLE
Add example contracts showcase

### PR DESCRIPTION
**Pull request by AurelienFT**
This PR adds a sub folder named `contracts_showcase` that contains contracts that are using objects from massa-as-sdk.
This PR contains :
- A README.md for user that want to run these contracts showcase.
- A CONTRIBUTING.md to ensure a standard in adding new contracts showcase
- A generic CI workflow that will tests all subfolders in the `contracts_showcase` folder.

### CODE REVIEW

File: contracts_showcase/accesscontrol/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "npx massa-as-compile",
+    "start": "npm run build && tsx interaction.ts"
+  },
+  "author": "",
+  "type": "module",
+  "license": "ISC",
+  "dependencies": {
+    "@massalabs/massa-as-sdk": "2.5.5-dev.20240402084724",
```
[the remark this answers is not in the document]
Comment (AurelienFT) [reply contracts_showcase/accesscontrol/package.json:14]: When I try you change I get 
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @massalabs/massa-as-sdk@2.5.5-dev.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in: /Users/aurelienfoucault/.npm/_logs/2024-04-25T08_29_12_332Z-debug-0.log
```

[the remark this answers is not in the document]
Comment (Thykof) [reply contracts_showcase/accesscontrol/package.json:14]: maybe you have `@massalabs/massa-as-sdk@2.5.5-dev.` but you should write `@massalabs/massa-as-sdk@2.5.5-dev` in the package.json

[the remark this answers is not in the document]
Comment (AurelienFT) [reply contracts_showcase/accesscontrol/package.json:14]: I tried this change as you asked and I add the error I posted above.

[the remark this answers is not in the document]
Comment (Thykof) [reply contracts_showcase/accesscontrol/package.json:14]: ok what you should have is "@massalabs/massa-as-sdk": "^2.5.5-dev",

the `^` is missing

File: contracts_showcase/accesscontrol/interaction.ts [hunk 2]
```diff
@@ -28,227 +15,112 @@ import * as dotenv from "dotenv";
...
-        .addString(userAccount.address!), // arguments for deployment
-    } as ISCData,
-    // Additional smart contracts can be added here for deployment
-  ],
-  chainId, // chain ID
-  fees, // fees for deployment
-  maxGas, // maximum gas for deployment
-  false // wait for the deployment finality
+const decimals = 10 ** 9;
+const apiUrl = "https://buildnet.massa.net/api/v2";
```
Comment (Thykof) [inline contracts_showcase/accesscontrol/interaction.ts:57]: I think this is exported by massa web3

File: contracts_showcase/accesscontrol/package.json [hunk 3]
```diff
@@ -11,10 +11,12 @@
   "type": "module",
   "license": "ISC",
   "dependencies": {
-    "@massalabs/massa-as-sdk": "^2.5.5-dev",
     "@massalabs/as-types": "^2.0.0",
+    "@massalabs/massa-as-sdk": "^2.5.5-dev",
     "@massalabs/massa-sc-deployer": "^1.3.0",
-    "@massalabs/massa-web3": "^3.0.2",
+    "@massalabs/massa-web3": "4.0.0-next.20240527122315",
```
Comment (Thykof) [inline contracts_showcase/accesscontrol/package.json:17]:
```suggestion
    "@massalabs/massa-web3": "4.0.0-next",
```

then run npm install

**Pull request closed without merging**